### PR TITLE
Create library function to apply site offset and use it in Activity Log

### PR DESCRIPTION
**Pull request by jsnajdr**
Creating or adjusting a moment.js object to the timezone of a particular site is a common activity that deserves a library function:
```js
import { applySiteOffset } from 'lib/site/timezone';

const timezone = site.options.timezone;
const gmtOffset = site.options.gmt_offset;
const datetime = applySiteOffset( '2018-03-18T17:04:00Z', { timezone, gmtOffset } );
```
creates a `datetime` object whose timezone/offset (whichever is defined) is set to what is set on the `site`.

This PR creates that library function and uses it in the Activity Log and the `clone-site` signup flow code.

One of the steps toward removing `moment-timezone` from `i18n-calypso`.

**Related changes:**
- use `withLocalizedMoment` or `useLocalizedMoment` for component that do date formatting with `moment`
- change some `moment.utc` calls to simple `moment`s. UTC parsing is not needed if the moment is immediately converted to another timezone.
- `ActivityLogAggregatedItem` uses `getSiteSlug` instead of the more generic `getSite`
- `ClonePointStep` never fetched the `timezone` and `gmtOffset` prop from the selected site, so timezone adjusting didn't work.
- more complex HOC chains benefit from `flowRight`/`compose`
- few ESLint warnings (unneeded `!!`) got fixed

**How to test:**
Go through your site's activity log and verify the timestamps are correct. And that expanding an "aggregated" item by clicking on it shows the correct expanded items.

I don't know how to easily test the `clone-site`. @rcoll might help with that?

### CODE REVIEW

**Comment by jsnajdr** [comment]
Found some `clone-site` testing instructions here: #25165. Still not easy 🏋️

File: client/lib/site/timezone.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ * @typedef OffsetParams
+ * @property {?string} timezone  Timezone representation to apply.
+ * @property {?string} gmtOffset Offset to apply if timezone isn't supplied.
+ */
+
+/**
+ * Accepts (optional) timezone and offset and applies one to the provided date, preferring the
+ * timezone. If neither are provided, creates a default moment.js object in local timezone.
+ *
+ * @param  {MomentInput}  input Valid input for moment (string, timestamp, moment.js object)
```
Comment (blowery) [inline client/lib/site/timezone.js]: is this typedef'd somewhere?

Comment (blowery) [reply client/lib/site/timezone.js]: https://github.com/moment/moment/blob/develop/moment.d.ts#L402

**Review by blowery — APPROVED** [review]
Code LGTM, though I'd love someone who worked on this to chime in.

**Review by gititon — APPROVED** [review]
LGTM

**Pull request merged**